### ISSUE TITLE
[Bug]: Nested routes under an id with a trailing `=` fail to match

### ISSUE THREAD

**Issue by mikeldking** [remix-run/react-router#8288]
### What version of React Router are you using?

6.0.0

### Steps to Reproduce

Create a nested route
```jsx
<Routes>
   <Route path="/dashboard/:dashboardId" element={<DashboardPage />}>
      <Route path="edit" element={<EditDasbhboard />}>
   </Route>
</Routes>
```

If the `:dahboardId` contains an equals sign at the end (e.x. a relay Id like `RGFzaGJvYXJkOjE2MzU=`), the generated regex `^\/dashboards\/([^\/]+)(?:\b|$)`  fails to capture the `=` at the end when the pathname is `/dashboards/RGFzaGJvYXJkOjE2MzU=/edit`

### Expected Behavior

The generated regex that matches the `:dashboardId` should capture all the way up to the next forward slash.

### Actual Behavior

`pathname.match` matches up to the `=` (`/dashboards/RGFzaGJvYXJkOjE2MzU`) but fails to capture the trailing `=`, causing the router to say that there are no matches for the nested route.

**Comment by mikeldking** [comment]
Here is an example of a failing test for a base64 encoded id (e.x. the output of `graphql-relay`'s `toGlobalId(`User`, `1`)`)

**Comment by mjackson** [comment]
Thanks for the bug report. I think the correct regex should probably be `^\/dashboards\/([^\/]+)(?:\/|$)` (look for the next / instead of a word boundary) but I'd have to test. Care to submit a PR?

**Comment by mikeldking** [comment]
@mjackson Thank you so much for responding and giving me an opportunity to contribute! 
Here is my PR: https://github.com/remix-run/react-router/pull/8291

Slight modification to the regex you posted above. Passes all previous unit tests plus new ones for `base64` encoded ids with trailing `=`s

**Comment by timdorr** [comment]
This should be fixed in 6.1.0.
